### PR TITLE
Initial block

### PR DESCRIPTION
**Pull request by hansmosh**

### CODE REVIEW

**Comment by hansmosh** [comment]
I didn't use the enrich signals mixin because it seemed simple enough without it.

File: file_reader_block.py [hunk 1]
```diff
@@ -10,6 +12,26 @@ class FileReader(Block):
...
+                self._logger.exception('Failed to evaluate file location')
+                continue
+            try:
+                self._logger.debug('Opening file: {}'.format(file))
+                with open(file) as openfile:
+                    self._logger.debug('Reading from file: {}'.format(file))
+                    setattr(signal, self.contents_attr, openfile.read())
+                    setattr(signal, self.file_attr, file)
+            except:
+                self._logger.exception('Failed to open/read file'.format(file))
```
Comment (mattdodge) [inline file_reader_block.py]: Looks like you're missing a `'{}'` in the string.

File: file_reader_block.py [hunk 2]
```diff
@@ -10,6 +12,26 @@ class FileReader(Block):
     Input signal trigger a read from a file. The contents of that file are
     placed on the notified signal in a configurable attribute.
     """
+    file = ExpressionProperty(title='File', default='/tmp/file.txt')
+    file_attr = StringProperty(title='File Field', default='file')
+    contents_attr = StringProperty(title='File Contents Field',
```
Comment (mattdodge) [inline file_reader_block.py:20]: Any reason we're not using the EnrichSignals mixin to do this?

File: file_reader_block.py [hunk 3]
```diff
@@ -10,6 +12,26 @@ class FileReader(Block):
...
-        pass
+        for signal in signals:
+            try:
+                file = self.file(signal)
+            except:
+                self._logger.exception('Failed to evaluate file location')
+                continue
+            try:
+                self._logger.debug('Opening file: {}'.format(file))
+                with open(file) as openfile:
```
Comment (mattdodge) [inline file_reader_block.py:35]: We have a function `NIOEnvironment.get_path(filename)` that will allow you to specify either a path relative to your project or an absolute path. That might be better here, I can see wanting to read from a project file

File: README.md [hunk 4]
```diff
@@ -1,24 +1,31 @@
...
+
+Input signal trigger a read from a file. The contents of that file are placed on the notified signal in a configurable attribute.
 
 Properties
---------------
-None
+----------
+
+-   **File** (expr): Location of file to read from. Path must be absolute.
+-   **File Field** (str): New attribute name to add to Signal that contains the **File**.
```
Comment (mattdodge) [inline README.md]: I think this is actually the filename, rather than the actual file object, is that right? Should clarify here

**Pull request merged**